frontend/app/assets/javascripts/app.js

```diff
@@ -0,0 +1,210 @@
+var myApp = angular.module('myApp', [ 'ngRoute', 'highcharts-ng' ]);
+
+myApp.controller('MyController', [ '$scope', function($scope) {
+  $scope.greeting = 'it works';
+  $scope.currencies = [];
+
+  // put code for highchart here
+  var myChart; // global variable to access chart
+
+  $(function() {
+
+    Highcharts.setOptions({
+      global : {
+        useUTC : false
+      }
+    });
+
+    // Create the chart
+    // $('#container').highcharts('StockChart', {
+    myChart = new Highcharts.StockChart({
+      chart : {
+        renderTo : 'container'// ,
+      // events : {
+      // load : function () {
+      //
+      // // set up the updating of the chart each second
+      // var series = this.series[0];
+      // setInterval(function () {
+      // var x = (new Date()).getTime(), // current time
+      // y = Math.round(Math.random() * 100);
+      // series.addPoint([x, y], true, true);
+      // }, 1000);
+      // }
+      // }
+      },
+
+      rangeSelector : {
+        buttons : [ {
+          count : 1,
+          type : 'minute',
+          text : '1M'
+        }, {
+          count : 5,
+          type : 'minute',
+          text : '5M'
+        }, {
+          type : 'all',
+          text : 'All'
+        } ],
+        inputEnabled : false,
+        selected : 0
+      },
+
+      title : {
+        text : 'Live stock data'
+      },
+
+      exporting : {
+        enabled : false
+      },
+
+      series : (function() {
+        // generate 10 series with name "i" and empty data
+        var seriesTmp = [];
+
+        for (var i = 0; i < 10; ++i) {
+          seriesTmp.push({
+            name : i.toString(),
+            data : [ 0, 0 ]
+          });
+        }
+        return seriesTmp;
+      }())
+    });
+
+    // $('#button').click(function () {
+    // var chart = $('#container').highcharts(),
+    // series = chart.series[0];
+    // if (series.visible) {
+    // series.hide();
+    // } else {
+    // series.show();
+    // }
+    // });
+
+  });
+
+  function toggleGraph(id) {
+    // console.log(this);
+    var button_id = id.split("_")[1];
+    series = myChart.series[button_id];
+    if (series.visible) {
+      series.hide();
+    } else {
+      series.show();
+    }
+  }
+
+  var ws = new WebSocket("ws://localhost:9000/wsTest");
+  ws.onmessage = function(event) {
+    // console.log("WS received TrueFx quotes:");
+
+    var quotes = JSON.parse(event.data).quotes;
+    // console.log(dataParsed);
+
+    var fill = false;
+    // initialize symbol names
+    if ($scope.currencies.length < 10) {
+      fill = true;
+    }
+
+    var x = (new Date()).getTime(); // current time for debugging (fetcher
```

Comment:
This will break if we speed up quote sending (e.g. when working with historical data at 2x speed).